.github/workflows/release.yml

```diff
@@ -2,47 +2,143 @@ name: Release
 
 on:
   workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+    types:
+      - closed
 
 jobs:
+
+  wheels_on_PR:
+    if: ${{ github.event.pull_request.merged == true }}
+    name: ${{ matrix.py }} ${{ matrix.os }} ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        arch: [native]
+        py: [cp310]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Setup gfortran for Linux
+        uses: awvwgk/setup-fortran@main
+        id: setup-fortran
+        with:
+          compiler: gcc
+          version: 11
+
+      - name: cibuildwheel on Linux
+        uses: pypa/cibuildwheel@v2.11.2
+        env:
+          CIBW_BUILD: ${{ matrix.py }}-*
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_ENVIRONMENT_PASS_LINUX: CI
+
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+
+
+
   wheels:
+    if: |
+      ${{ github.event_name == 'workflow_dispatch' ||
+      (github.ref == 'refs/heads/main' && startsWith(github.ref, 'refs/tags/v')) }}
     name: ${{ matrix.py }} ${{ matrix.os }} ${{ matrix.arch }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        arch: [auto, aarch64]
-        py: [cp37, cp38, cp39, cp310]
-        include:
-          - os: macos-latest
-            py: cp38
-            arch: universal2
-          - os: macos-latest
-            py: cp39
-            arch: universal2
-          - os: macos-latest
-            py: cp310
-            arch: universal2
+        arch: [native, arm64]
+        py: [cp38, cp39, cp310, cp311]
         exclude:
-          - os: windows-latest
-            arch: aarch64
-          - os: macos-latest
-            arch: aarch64
-    env:
-      CIBW_BUILD: ${{ matrix.py }}-*
-      CIBW_ARCHS_LINUX: ${{ matrix.arch }}
+            - os: macos-latest
+              arch: arm64
+              py: cp38
+            - os: ubuntu-latest
+              arch: arm64
+            - os: windows-latest
+              arch: arm64
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: true
 
-      - if: ${{ matrix.arch == 'aarch64' }}
-        uses: docker/setup-qemu-action@v1
+      - if: ${{ matrix.os == 'macos-latest' &&  matrix.arch == 'native'}}
```

Comment:
Are you sure that you can remove this? This is required to build arch64 wheels, except on macos, where we can build universal wheels that work on x86 and arch64.

Reply:
I removed it, because it never used by now. I will add Linux aarch64 when other systems start to work.

Reply:
Then keep it and just modify the matrix so that arch64 is not built.

Reply:
Ok. I restored these lines.